Source/Core/Core/IOS/USB/Host.cpp

```diff
@@ -56,7 +64,50 @@ std::optional<IPCReply> USBHost::Open(const OpenRequest& request)
 std::string USBHost::GetDeviceNameFromVIDPID(u16 vid, u16 pid)
 {
   std::string device_name;
-#ifdef __LIBUSB__
+#ifdef _WIN32
+  const std::wstring filter = fmt::format(L"VID_{:04X}&PID_{:04X}", vid, pid);
+
+  HDEVINFO dev_info =
+      SetupDiGetClassDevs(nullptr, nullptr, nullptr, DIGCF_PRESENT | DIGCF_ALLCLASSES);
+  if (dev_info == INVALID_HANDLE_VALUE)
+    return device_name;
+
+  SP_DEVINFO_DATA dev_info_data;
+  dev_info_data.cbSize = sizeof(SP_DEVINFO_DATA);
+
+  for (DWORD i = 0; SetupDiEnumDeviceInfo(dev_info, i, &dev_info_data); ++i)
+  {
+    TCHAR instance_id[MAX_DEVICE_ID_LEN];
+    if (CM_Get_Device_ID(dev_info_data.DevInst, instance_id, MAX_DEVICE_ID_LEN, 0) != CR_SUCCESS)
+      continue;
+
+    const std::wstring_view id_wstr(instance_id);
+    if (id_wstr.find(filter) == std::wstring::npos)
```

Comment:
I'm not familiar with the exact format of the "device ID", but is a simple substring search for `VID_XXXX&PID_XXX` a foolproof test?